src/dune_lang/string_with_vars.ml

```diff
@@ -132,17 +132,26 @@ let concat_rev = function
 ;;
 
 module Mode = struct
-  type _ t =
-    | Single : Value.t t
-    | Many : Value.t list t
-    | At_least_one : (Value.t * Value.t list) t
+  type (_, _) t =
+    | Single : (Value.Deferred.t, Value.t) t
+    | Many : (Value.Deferred.t list, Value.t list) t
```

Comment:
Is it ever possible to expand into a `Value.Deferred.t list` anywhere?

Reply:
Yep, see the `Action_expander.slang_expander` function